util/genrest/goviewcreator.go

```diff
@@ -18,47 +18,130 @@ import (
+		file.P("")
 
 		importStrings := make([]string, 0, len(service.Imports))
 		for _, spec := range service.Imports {
-			importStrings = append(importStrings, fmt.Sprintf("%s: %q", spec.Name, spec.Path))
+			importStrings = append(importStrings, fmt.Sprintf("%s %q", spec.Name, spec.Path))
 		}
 		sort.Strings(importStrings)
-		file.P(strings.Join(importStrings, "\n"))
 
-		file.P("\n--- Handlers ----------\n")
+		file.P("import (")
+		file.P(`  "context"`)
+		file.P(`  "net/http"`)
+		file.P("")
+		// TODO: Properly deal with imports once we actually use them in the code.
+		// file.P("  // %s", strings.Join(importStrings, "\n  // "))
+
+		// TODO: Get the following import from the data model, rather than hard-coding, once we deal with all imports.
+		file.P(`	  genprotopb "github.com/googleapis/gapic-showcase/server/genproto"`)
+
+		// TODO: Investigate why "google.golang.org/protobuf" doesn't work below
+		file.P(`  "github.com/golang/protobuf/jsonpb"`)
+		file.P(")")
+		file.P("")
 
 		for _, handler := range service.Handlers {
 			handlerName := namer.Get("Handle" + handler.GoMethod)
-			file.P("//                For :%s", handler.URIPattern)
-			file.P("// Match URIs of form :%s", matchingPath(handler.PathTemplate))
-			file.P("func %s(w http.ResponseWriter, r *http.Request) {", handlerName)
-			file.P("  // serialize from r to: var %s %s.%s", handler.RequestVariable, handler.RequestTypePackage, handler.RequestType)
-			file.P("  %s := %s(%s)", handler.ResponseVariable, handler.GoMethod, handler.RequestVariable)
-			file.P("  // serialize back to w\n")
+			pathMatch := matchingPath(handler.PathTemplate)
+			registered = append(registered, &registeredHandler{pathMatch, handlerName})
+
+			file.P("")
+			file.P("// %s translates REST requests/responses on the wire to internal proto messages for %s", handlerName, handler.GoMethod)
+			file.P("//    Generated for HTTP binding pattern: %s", handler.URIPattern)
+			file.P("//         This matches URIs of the form: %s", pathMatch)
+			file.P("func (backend *RESTBackend) %s(w http.ResponseWriter, r *http.Request) {", handlerName)
+			file.P(`  backend.StdLog.Printf("Received request matching '%s': %%q", r.URL)`, handler.URIPattern)
+			if handler.StreamingClient || handler.StreamingServer {
+				file.P(`  w.Write([]byte("ERROR: not implementing streaming methods yet"))`)
+				file.P("}")
+				continue
+			}
+
+			file.P("")
+			file.P("  var %s *%s.%s", handler.RequestVariable, handler.RequestTypePackage, handler.RequestType)
+			file.P("  // TODO: Populate %s with parameters from HTTP request", handler.RequestVariable)
+			file.P("")
+			// TODO: In the future, we may want to redirect all REST-endpoint requests to the gRPC endpoint so that the gRPC-registered observers get invoked.
+			file.P("  %s, err := backend.%sServer.%s(context.Background(), %s)", handler.ResponseVariable, service.ShortName, handler.GoMethod, handler.RequestVariable)
+			file.P("  if err != nil {")
```

Comment:
Just curious, what's the motivation for adding this literal block line-by-line rather than the entire subblock at once?

Reply:
Primarily consistency, adhering to the convention used in gapic-generator-go and, I think, gapic-showcase.

You bring up a good point, though, that for multi-line literals we could use something like [dedent](https://github.com/lithammer/dedent)

Might be worth considering in the future; for now, I'm inclined to not add more changes.